vm/actor/src/builtin/miner/mod.rs

```diff
@@ -2248,65 +3010,151 @@ where
     let unsealed_cid: Cid = ret.deserialize()?;
     Ok(unsealed_cid)
 }
+
+fn request_deal_weight<BS, RT>(
+    rt: &mut RT,
+    deal_ids: &[DealID],
+    sector_start: ChainEpoch,
+    sector_expiry: ChainEpoch,
+) -> Result<VerifyDealsForActivationReturn, ActorError>
+where
+    BS: BlockStore,
+    RT: Runtime<BS>,
+{
+    let serialized = rt.send(
+        *STORAGE_MARKET_ACTOR_ADDR,
+        MarketMethod::VerifyDealsForActivation as u64,
+        Serialized::serialize(VerifyDealsForActivationParamsRef {
+            deal_ids,
+            sector_start,
+            sector_expiry,
+        })?,
+        TokenAmount::zero(),
+    )?;
+
+    Ok(serialized.deserialize().unwrap())
```

Comment:
dont need unwrap, can just `?` as it will convert to a serialize error (I see you are trying to match their behaviour, but when they panic with a serialize error, they recover into the ActorError/ExitCode needed

Reply:
Ah, very good to know